src/dune_rules/scope.ml

```diff
@@ -473,15 +460,22 @@ module DB = struct
              | Some lib ->
                let name = Package.name pkg in
                (name, Lib_entry.Library (Lib.Local.of_lib_exn lib)) :: acc)
-          | Library.T { visibility = Public pub; _ } ->
-            let+ lib = Lib.DB.find public_libs (Public_lib.name pub) in
+          | Library.T { visibility = Public pub; enabled_if; _ } ->
+            let* lib = Lib.DB.find public_libs (Public_lib.name pub) in
```

Comment:
You should use the id rather than the name for this check.

Reply:
I tried `find_by_lib_id` but I got a crash about adding a duplicate name in https://github.com/ocaml/dune/blob/main/src/dune_rules/install_rules.ml#L697

Reply:
I'm not sure what that means yet.

Reply:
Ah, it's likely because it's finding the right lib now.